RELEASE/scripts/autoscend/auto_util.ash

```diff
@@ -889,22 +881,53 @@ boolean forbidFamChange()
 	{
 		return false;
 	}
+	
+	// if you reached this line, then it means that auto_100familiar is set to some specific familiar.
 	return true;
 }
 
-boolean forbidFamChange(familiar thisOne)
+boolean forbidFamChange()
 {
-	if(forbidFamChange())
+	// answers the question "am I forbidden to change familiar?"
+	// an answer of true means you cannot change familiar, either due to path or due to being 100% run.
+	// an answer of false means you are allowed to change familiar
+	
+	//paths in which you cannot use a familiar as a familiar (in pokefam familiars are used as something else)
+	if($strings[Actually Ed the Undying, Avatar of Boris, Avatar of Jarlsberg, Avatar of Sneaky Pete, License to Adventure, Pocket Familiars, Dark Gyffte] contains auto_my_path())
 	{
-		if(get_property("auto_100familiar") == thisOne)
-		{
-			return false;
-		}
 		return true;
 	}
-	return false;
+	
+	// If you are not in a specific path that forbids familiars, then the question is whether or not you are in a 100% run. If you are in such a run then changing familiar is forbidden.
+	return is100FamRun();
 }
 
+boolean forbidFamChange(familiar target)
```

Comment:
I'm not particularly enamoured with the name of this function mostly due to the double negative as you explain in the comment.
Something like `boolean canChangeToFamiliar(familiar target)` (which returns true when you can change and false when you can't) should be more obvious.

Reply:
I am not a fan of the double negative, but it is what the function actually does.
Changing it to canChangeToFamiliar means we need to find every instance in which it is being used and flip it with a ! or removing a ! if it already has one.
It is doable, but far more likely to introduce errors.

It is also worth noting that the function name change was done in a previous commit. Here I only changed "familiar thisOne" into "familiar target" to make it clearer that this is a "change to familiar X"

If everyone agrees then I don't mind making anew PR where I switch
`forbidFamChange()`
into
`canChangeToFamiliar()`
and go through every single instance in the code where it occurs and either add or subtract a !

Reply:
ok, I talked to jeparo and he is also in favor of the change. And thinking about it some more I like it too...
So, after this PR goes through I will immediately make a new PR to flip `forbidFamChange(target)` into `canChangeToFamiliar(target)`
and `forbidFamChange()` into `canChangeFamiliar()`